parachains/runtimes/assets/asset-hub-westend/src/xcm_config.rs

```diff
@@ -125,6 +127,17 @@ pub type ForeignAssetsConvertedConcreteId = assets_common::ForeignAssetsConverte
 	Balance,
 >;
 
+/// `AssetId/Balance` converter for pay by swap
+pub type LocalAndForeignAssetsConvertedConcreteId = assets_common::ForeignAssetsConvertedConcreteId<
```

Comment:
I am not sure if this is good way and if it covers everything: Local and Foreign... it is kind of hacky.

what was wrong with two `SwapFirstAssetTrader` instance configured with `Assets` and `ForeignAssets` with reused verified converters?
```
SwapFirstAssetTrader<Assets, TrustBackedAssetsConvertedConcreteId>
SwapFirstAssetTrader<ForeignAssets, ForeignAssetsConvertedConcreteId>
```

xcm uses everywhere tuple implementation, e.g. check AssetTransactors, it is much clear approach,
maybe in future we would like to add another pallet_assets instance, then what? `LocalAndForeignAndWhateverAssetsConvertedConcreteId + LocalAndForeignAndWhateverAssets` ?
I guess easier/nicer is to add another `SwapFirstAssetTrader`

Reply:
Yeah, I agree. I think a lot of the problem originates in the `SignedExtension` for fee payment. The `LocalOrForeign` struct basically serves as the router, to figure out which instance of the Assets pallet the input refers to. But I don't think we need to do that here.

Reply:
I can try using two SwapFirstAssetTrader instances for now and if and when LocalAndForeignAssets gets conformance tests, we can switch to that

Reply:
well, I would not switch back to `LocalAndForeignAssets` here :),
as Joe said `LocalAndForeignAssets` is ok/needed for `SignedExtension`, but here is much clear to have multiple `SwapFirstAssetTrader`

Reply:
For the `Assets` one, how do I fulfill the bound in `SwapFirstAssetTrader` that is: `	MultiLocation: From<ConcreteAssets::AssetId> + Into<T::MultiAssetId>`

[Here](https://github.com/paritytech/cumulus/pull/2854/files#diff-7bc584d0935b865ede04e0e6d9a9f66721282e96df9a54c64826523a3759e2fbR318)'s the constraint

The question is explicitly, what would be the right `MatchedConvertedConcreteId` (asset matcher) type to pass in here?

Reply:
@PatricioNapoli 
for `Assets` -> `xcm_config::TrustBackedAssetsConvertedConcreteId`
for `ForeignAssets` -> `xcm_config::ForeignAssetsConvertedConcreteId`

and for that constraint, I tried it (does not compile because of XcmContext), but something like this should work:
https://github.com/paritytech/cumulus/pull/2955